lib/clearance/authentication.rb

```diff
@@ -62,8 +62,16 @@ def current_user=(user)
     #
     # For an example of how clearance uses this internally, see
     # {SessionsController#create}.
+    #
+    # Signing in will also regenerate the CSRF token for the current session,
+    # provided {Configuration#rotate_csrf_token_on_sign_in} is set.
     def sign_in(user, &block)
-      clearance_session.sign_in user, &block
+      clearance_session.sign_in(user, &block)
+
+      if signed_in? && Clearance.configuration.rotate_csrf_on_sign_in?
```

Comment:
Does it make more sense to do this in `clearance_session`? Currently, this method is a pretty straightforward delegator. Adding logic here means that you need both this module and the session object to actually sign in.